lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1002,6 +1005,7 @@ def __init__(
         self.relationship_name = relationship_name
         self.certificate_requests = certificate_requests
         self.mode = mode
+        self._private_key = private_key
```

Comment:
We should probably have some input validation to make sure to quickly inform the user if the key is not valid.

Reply:
I added a validation function with the following logic:
- Make sure it is RSA, we check that in other functions so we don't want the user to provide a private key that used a different algorithm (although that shouldn't happen coming from TLS V3)
- Make sure whatever is passed is a valid key and not random string.
- Added a security check for the key to be at least 2048 bits. (This a breaking change I know but I think it is an important security enhancment)